trillian/ctfe/cert_quota.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctfe
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// CertificateQuotaUserPrefix is prepended to all User quota ids association
+// with intermediate certificates.
+const CertificateQuotaUserPrefix = "@intermediate"
+
+// QuotaUserForCert returns a User quota id string for the passed in
+// certificate.
+// This is intended to be used for quota limiting by intermediate certificates,
+// but the function does not enforce anything about the passed in cert.
+//
+// Format returned is:
+//   "CertificateQuotaUserPrefix Subject SHA256_hex(SubjectPublicKeyInfo[0:5])"
```

Comment:
`SHA256_hex(SubjectPublicKeyInfo)[0:5]`  I think (different end bracket order).

Reply:
They're both wrong :)
Yours 'cos that'll be 5 ASCII bytes of hex with my imaginary, and mine because it'll be the wrong hash...
fixed by separating the sha256 and hexification